langchain/llms/huggingface_hub.py

```diff
@@ -0,0 +1,102 @@
+"""Wrapper around HuggingFace APIs."""
+import os
+from typing import Any, Dict, List, Mapping, Optional
+
+from pydantic import BaseModel, Extra, root_validator
+
+from langchain.llms.base import LLM
+from langchain.llms.utils import enforce_stop_tokens
+
+DEFAULT_REPO_ID = "gpt2"
+
+
+class HuggingFaceHub(BaseModel, LLM):
+    """Wrapper around HuggingFaceHub  models.
+
+    To use, you should have the ``huggingface_hub`` python package installed, and the
+    environment variable ``HUGGINGFACEHUB_API_TOKEN`` set with your API token.
+
+    Only supports task `text-generation` for now.
+
+    Example:
+        .. code-block:: python
+
+            from langchain import HuggingFace
+            hf = HuggingFace(model="text-davinci-002")
+    """
+
+    client: Any  #: :meta private:
+    repo_id: str = DEFAULT_REPO_ID
+    """Model name to use."""
+    temperature: float = 0.7
+    """What sampling temperature to use."""
+    max_new_tokens: int = 200
+    """The maximum number of tokens to generate in the completion."""
+    top_p: int = 1
+    """Total probability mass of tokens to consider at each step."""
+    num_return_sequences: int = 1
+    """How many completions to generate for each prompt."""
+
+    class Config:
+        """Configuration for this pydantic object."""
+
+        extra = Extra.forbid
+
+    @root_validator()
+    def validate_environment(cls, values: Dict) -> Dict:
+        """Validate that api key and python package exists in environment."""
+        if "HUGGINGFACEHUB_API_TOKEN" not in os.environ:
+            raise ValueError(
+                "Did not find HuggingFace API token, please add an environment variable"
+                " `HUGGINGFACEHUB_API_TOKEN` which contains it."
+            )
+        try:
+            from huggingface_hub.inference_api import InferenceApi
+
+            repo_id = values.get("repo_id", DEFAULT_REPO_ID)
+            values["client"] = InferenceApi(
+                repo_id=repo_id,
+                token=os.environ["HUGGINGFACEHUB_API_TOKEN"],
```

Comment:
These parameters seem like nice ones to include as init args with optional env args as backups.

Also the relationship feels a bit convoluted at first glance

Reply:
what in particular seems convoluted?

Reply:
I think the main thing is that the client is created in a validation function - but that seems out of scope of this PR given the existing structure

Reply:
ah yeah i agree. is a bit weird. may look into __post_init__